include/xgboost/multi_target_tree_model.h

```diff
@@ -19,12 +19,50 @@
 
 namespace xgboost {
 struct TreeParam;
+/**
+ * @brief A view to the @MultiTargetTree suitable for both host and device..
+ */
+struct MultiTargetTreeView {
+  static bst_node_t constexpr InvalidNodeId() { return -1; }
+
+  common::Span<bst_node_t const> left;
```

Comment:
Are these spans all the same size? I wonder if it's worth specifying a template parameter

```
template<size_t N = std::dynamic_extent>
struct MultiTargetTreeView {

  common::Span<bst_node_t const, N> left;
  common::Span<bst_node_t const, N> right;
  ...
```

Reply:
Yeah, they have the same size. We can replace the spans with pointera if needed.

Reply:
I changed them into pointers, the extra template parameter won't be needed.